public/components/layout.js

```diff
@@ -239,64 +240,222 @@ class PointPicker extends HTMLElement {
+        localStorage.removeItem('coordinates')
+        this.pointList = []
+        this.markerGroup.clearLayers()
+        this.clearMapLayers()
+    }
+
+    /**
+     * Clears all previously drawn shapes on the preview map.
+     * @returns {None}
+     */
+    clearMapLayers() {
+        const layerTypes = {
+            "LineString": L.Polyline,
+            "Polygon": L.Polygon
+        };
+        
+        this.previewMap.eachLayer(layer => {
+            for (const geomType in layerTypes) {
+                if (layer instanceof layerTypes[geomType]) {
+                    this.previewMap.removeLayer(layer);
+                    break;
+                }
+            }
+        });
+    }
+
+    /**
+     * Highlights the geometry type selected by the user
+     * @param {String} newGeomChoice A string indicated the geometry type selected by the user 
+     * @returns {None} 
+    */
+    highlightGeomType(newGeomChoice) {
+        if (newGeomChoice === "Point") {
+            pointBtn.style.border = "3px solid black";
+            polygonBtn.style.border = "0px solid black";
+            polylineBtn.style.border = "0px solid black";
+        } else if (newGeomChoice === "LineString") {
+            pointBtn.style.border = "0px solid black";
+            polygonBtn.style.border = "0px solid black";
+            polylineBtn.style.border = "3px solid black";
+        } else if (newGeomChoice === "Polygon") {
+            pointBtn.style.border = "0px solid black";
+            polygonBtn.style.border = "3px solid black";
+            polylineBtn.style.border = "0px solid black";
+        }
     }
 
     /**
      * Take the coordinates provided by the user and turn them into GeoJSON
-     * @param none
-     * @return none
+     * @param {none}
+     * @return {none}
      */
     confirmCoordinates() {
-        let lat = parseInt(leafLat.value * 1000000) / 1000000
-        let long = parseInt(leafLong.value * 1000000) / 1000000
         let geo = {}
-        if (lat && long) {
-            geo.type = "Point"
-            geo.coordinates = [long, lat]
+        const geometry_type = localStorage.getItem('geometryType') ?? 'Point'
+        const coords = JSON.parse(localStorage.getItem('coordinates'))
+        geo.type = geometry_type
+        geo.coordinates = []
+        for (let index = 0; index < coords.length; index += 2) {
+            let long = parseInt(coords[index] * 1000000) / 1000000
+            let lat = parseInt(coords[index+1] * 1000000) / 1000000
+            geo.coordinates.push([long, lat])
         }
-        else {
-            alert("Supply both a latitude and a longitude")
-            return false
+        switch(geometry_type){
+            case 'Point':
+                geo.coordinates = geo.coordinates[0]
+                break
+            case 'LineString':
+                geo.coordinates = geo.coordinates
+                break
+            case 'Polygon':
```

Comment:
Clean this up a bit

Here is a one liner that accomplishes the same thing (I think, just typed it out I did not test it)
```javascript
geo.coordinates = [...geo.coordinates, ...[geo.coordinates[0]]]
```

Reply:
The requested changes have been added in new commits